igvc_platform/src/imu/main.cpp

```diff
@@ -115,128 +132,136 @@ int main(int argc, char** argv)
       ROS_ERROR_STREAM("error got unknown message type " << response.messageType);
     }
 
+    // Get accelerometer measurements.
     MultiAxisSensor accel_msg;
     MultiAxisSensor accel_raw_msg;
 
     ret = freespace_util_getAccNoGravity(&response.motionEngineOutput, &accel_msg);
+    if (ret != FREESPACE_SUCCESS)
+    {
+      ROS_ERROR_STREAM("failed to read acceleration no gravity." << ret);
+    }
+
     ret = freespace_util_getAcceleration(&response.motionEngineOutput, &accel_raw_msg);
     if (ret != FREESPACE_SUCCESS)
     {
-      ROS_ERROR_STREAM("failed to read acceleration " << ret);
+      ROS_ERROR_STREAM("failed to read acceleration." << ret);
     }
-    Eigen::Vector3d accel_vec(-accel_msg.x, accel_msg.y, accel_msg.z);
-    Eigen::Vector3d accel_raw_vec(-accel_raw_msg.x, accel_raw_msg.y, accel_raw_msg.z);
-    Eigen::Matrix3d T;
-    T << cos(-M_PI / 2), -sin(-M_PI / 2), 0, sin(-M_PI / 2), cos(-M_PI / 2), 0, 0, 0, 1;
-    Eigen::Vector3d rotated_accel = T * accel_vec;
-    Eigen::Vector3d rotated_raw_accel = T * accel_raw_vec;
+
+    Eigen::Vector3d accel_raw_vec(accel_raw_msg.x, accel_raw_msg.y, accel_raw_msg.z);
+    accel_raw_vec = correction_mat * accel_raw_vec;
+
+    Eigen::Vector3d accel_vec(accel_msg.x, accel_msg.y, accel_msg.z);
+    accel_vec = correction_mat * accel_vec;
 
     MultiAxisSensor orientation_msg;
     ret = freespace_util_getAngPos(&response.motionEngineOutput, &orientation_msg);
     if (ret != FREESPACE_SUCCESS)
     {
       ROS_ERROR_STREAM("failed to read angular position " << ret);
     }
+    tf::Quaternion quaternion_raw(orientation_msg.x, orientation_msg.y, orientation_msg.z, orientation_msg.w);
+    double roll, pitch, yaw;
+    tf::Matrix3x3(quaternion_raw).getRPY(roll, pitch, yaw);
+    ROS_INFO("Filtered heading: %f", yaw);
```

Comment:
maybe not print out to terminal. We have a launch file that will convert quat to r,p,y on the fly on util

Reply:
Done